platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
```

Comment:
Where did `list` go? That seems important. Something like:

`fn list(url: ResourceUrl) -> Result<Vec<String>>`

As for the behaviour of that, you could do it like POSIX `ls`, where a `list` call on a file resource would return the file itself, but on a directory, it would be the contents. Also worth thinking about whether it would always list recursively, or just immediate children. Former seems to suffice, but could be unnecessarily expensive.

Lastly, we currently have an implicit root directory, which you get by supplying an empty string here. Not sure that's a great design: POSIX has `.` for the current directory (which is the same as not supplying any directory to `ls`), and `/` for the root directory, but this stuff doesn't work for us, because we work with `polypod:` URLs, not POSIX paths. Maybe `list("polypod://")` would be that? Not sure. Maybe a constant for that then?

Reply:
`list` is replaced by `read` since they do the same thing. `list` reads the contents for the current feature stored under `FeatureFiles/FeatureName`.

But thinking about it, what should someone pass to read to get the behaviour of list? 🤔 TBD